low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -182,20 +183,35 @@ def process(self):
 
 
 class TestCmd(Cmd):
-    """Starts test based on plan (sanity | alerts)."""
+    """Starts test based on plan:
+    (sanity|alerts|self_primary|self_secondary)."""
 
     name = "test"
 
     def __init__(self, args):
         super().__init__(args)
 
     def validate(self):
-        # Common validator classes to check Cortx/system wide validator
-        pass
+        if not self.args:
+            default_plan = "self_primary"
+            self.args.insert(0, default_plan)
+        elif len(self.args) >= 1:
+            i=0
+            while i < len(self.args):
+                try:
+                    if self.args[i] in ["sanity", "alerts", "self_primary","self_secondary"]:
+                        break
+                except (IndexError, ValueError):
+                    raise SetupError(1, "Invalid plan type specified. Please check usage")
+                i+=1
+
+        result = PkgV().validate("rpms", "sspl-test")
+        if result == -1:
+            raise SetupError(1, "'sspl-test' rpm pkg not found.")
```

Comment:
also collect config value here, but dont fail if not given or validate even, as we dont use it, its mainly for future use.

Reply:
Done